File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(t) < len(s):
            return False
        elif len(t) == len(s):
            if s == t:
                return True
            else:
                return False
        sp = 0
        tp = 0
        
        while sp < len(s) and tp < len(t):
            if s[sp] == t[tp]:
                sp += 1
            tp += 1
        if sp == len(s):
            return True
        return False
            

File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int sp = 0;

        if(s.length() > t.length()) {
            return false;
        }
        if(s.isEmpty()) {
            return true;
        }

        for(int i = 0; i < t.length(); i++) {
            if(sp < s.length() && s.charAt(sp) == t.charAt(i)) {
                sp++;
            }
        }

        if(sp >= s.length()) {
            return true;
        }
        return false;
    }
}